packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx

```diff
@@ -99,10 +99,12 @@ export class SpinButton extends BaseComponent<ISpinButtonProps, ISpinButtonState
       value = String(Math.max(newProps.min as number, Math.min(newProps.max as number, Number(newProps.defaultValue))));
     }
 
-    this.setState({
-      value: value,
-      precision: newProps.precision || this.state.precision
-    });
+    if (newProps.value !== undefined) {
+      this.setState({
+        value: value,
+        precision: newProps.precision || this.state.precision
```

Comment:
In theory, precision should be updated even if value is undefined. (Also, the || check won't work right if the new precision is 0.)

Reply:
Actually precision shouldn't be in the state at all (since state shouldn't be used to cache calculations)... It would be more appropriate to remove precision from the state and replace all usage with calls to a function like this. The function is memoized so it will only run the calculation once per combination of precision and step values.

```ts
const _getPrecision = memoizeFunction((precision: number| undefined, step: number | undefined): number => {
  if (typeof precision === 'number') {
    return precision;
  }
  // Ensure that the autocalculated precision is not negative.
  return Math.max(calculatePrecision(step!), 0);
});
```

Reply:
done.